server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/DistributedHealthCheckIntegrationTest.java

```diff
@@ -0,0 +1,82 @@
+
+package org.apache.james.webadmin.integration.rabbitmq;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.james.CassandraExtension;
+import org.apache.james.CassandraRabbitMQJamesConfiguration;
+import org.apache.james.CassandraRabbitMQJamesServerMain;
+import org.apache.james.DockerOpenSearchExtension;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.JamesServerBuilder;
+import org.apache.james.JamesServerExtension;
+import org.apache.james.SearchConfiguration;
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.events.RabbitMQEventBusDeadLetterQueueHealthCheck;
+import org.apache.james.modules.AwsS3BlobStoreExtension;
+import org.apache.james.modules.RabbitMQExtension;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.modules.blobstore.BlobStoreConfiguration;
+import org.apache.james.queue.rabbitmq.RabbitMQMailQueueDeadLetterQueueHealthCheck;
+import org.apache.james.utils.GuiceProbe;
+import org.apache.james.webadmin.integration.HealthCheckProbe;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.google.inject.multibindings.Multibinder;
+
+class DistributedHealthCheckIntegrationTest {
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerBuilder<CassandraRabbitMQJamesConfiguration>(tmpDir ->
+        CassandraRabbitMQJamesConfiguration.builder()
+            .workingDirectory(tmpDir)
+            .configurationFromClasspath()
+            .blobStore(BlobStoreConfiguration.builder()
+                .s3()
+                .disableCache()
+                .deduplication()
+                .noCryptoConfig())
+            .searchConfiguration(SearchConfiguration.openSearch())
+            .build())
+        .extension(new DockerOpenSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .extension(new RabbitMQExtension())
+        .server(configuration -> CassandraRabbitMQJamesServerMain.createServer(configuration)
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, GuiceProbe.class).addBinding().to(HealthCheckProbe.class))
+            .overrideWith(new TestJMAPServerModule()))
+        .build();
+
+    @Test
+    void shouldContainRabbitMQDeadLetterQueueHealthChecks(GuiceJamesServer server) {
+        List<ComponentName> healthCheckComponentNames = server.getProbe(HealthCheckProbe.class).getHealthCheckSet()
+            .stream()
+            .map(HealthCheck::componentName)
+            .collect(Collectors.toList());
+
+        assertThat(healthCheckComponentNames).contains(RabbitMQEventBusDeadLetterQueueHealthCheck.COMPONENT_NAME,
+            RabbitMQMailQueueDeadLetterQueueHealthCheck.COMPONENT_NAME);
```

Comment:
Can't we list all healtch checks?

Reply:
Maybe specify hard coded names and specify raw strings?

It will help us catch breaking changes!